### DISCUSSION TITLE
Help! The onMount fires when $data from API hasn't yet returned into a writable store, shall I use a callback?

### DISCUSSION THREAD

**Discussion by federicomarcantognini** [sveltejs/kit#11208] [General]
When I use a writable store into the onMount in a component, the onMount fires when the data is not yet resolved from my API, therefore $data is still empty when onMount fires.

How to ensure that $data is populated from API response right into onMount?  A callback? 
Many thanks in advance and apologies for the silly question

**Comment by alexbjorlig** [comment]
> When I use a writable store into the onMount in a component, the onMount fires when the data is not yet resolved from my API, therefore $data is still empty when onMount fires.
> 
> How to ensure that $data is populated from API response right into onMount? A callback? Many thanks in advance and apologies for the silly question

Can you share code to show what you are trying to build - then it's much more easy to help answer your questions :)

**Comment by Garito** [comment]
Instead of using onMount, you can call the function when the data is updated like:
```javascript
$: $data, init();

const init = () => {
  if ($data) {
    // $data is not undefined here
  }
}
```
